File: Medii de Proiectare si Programare/TrackAndField-C#/TrackAndField/model/Track.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace TrackAndField.model
{
    class Track : Entity<string>
    {
        public string Name { get; set; }
        public int MinimumAge { get; set; }
        public int MaximumAge { get; set; }
        public Track(string name, int minimumAge, int maximumAge)
        {
            ID = name;
            Name = name;
            MinimumAge = minimumAge;
            MaximumAge = maximumAge;
        }
        public override string ToString()
        {
            return Name + " | min age: " + MinimumAge + " | max age: " + MaximumAge;
        }
    }
}

File: Medii de Proiectare si Programare/TrackAndField-Java/src/main/java/model/Track.java
package model;

public class Track extends Entity<String> implements Comparable {
    private String name;
    private int minimumAge;
    private int maximumAge;

    public Track(String name, int minimumAge, int maximumAge) {
        this.setId(name);
        this.name = name;
        this.minimumAge = minimumAge;
        this.maximumAge = maximumAge;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) { this.name = name; }

    public int getMinimumAge() {
        return minimumAge;
    }

    public void setMinimumAge(int minimumAge) {
        this.minimumAge = minimumAge;
    }

    public int getMaximumAge() {
        return maximumAge;
    }

    public void setMaximumAge(int maximumAge) {
        this.maximumAge = maximumAge;
    }

    @Override
    public int compareTo(Object o) {
        Track other = (Track)o;
        if(this.name.equals(other.name))
            return 0;
        if(this.name.compareTo(other.name) < 0)
            return -1;
        return 1;
    }

    @Override
    public String toString() {
        return this.name + " | min age: " + this.minimumAge + " | max age: " + this.maximumAge;
    }
}
